### PR TITLE
Mention oneDPL TPL

### LINKED ISSUE #494: oneDPL third-party library is not mentioned in the documentation

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #494.

### CODE REVIEW

File: docs/source/API/core/Macros.rst [hunk 1]
```diff
@@ -134,6 +134,8 @@ These defines give information about what third-party libaries Kokkos was compil
 +-------------------------------+-----------------------------------------------------------------------------------------------------------------------+
 | ``KOKKOS_ENABLE_LIBQUADMATH`` | Defined if Kokkos links to the `GCC Quad-Precision Math Library API <https://gcc.gnu.org/onlinedocs/libquadmath/>`_.  |
 +-------------------------------+-----------------------------------------------------------------------------------------------------------------------+
+| ``KOKKOS_ENABLE_ONEDPL``      | Defined if Kokkos links to the `oneDPL library <https://github.com/oneapi-src/oneDPL>`_ when using the SYCL backend.  |
++-------------------------------+-----------------------------------------------------------------------------------------------------------------------+
```
Comment (dalg24) [inline docs/source/API/core/Macros.rst:138]: Should we expand to say what it is used for and that things like `Kokkos::sort` might be less performant if not found?

It is fine if you want to do as a follow up.

Comment (masterleinad) [reply docs/source/API/core/Macros.rst]: [`d02c302` (#496)](https://github.com/kokkos/kokkos-core-wiki/pull/496/commits/d02c302d8856c12976718bca2e5396f95a48c848)

**Review by nmm0 — APPROVED** [review]
Looks good to me!

File: docs/source/keywords.rst [hunk 2]
```diff
@@ -225,6 +225,9 @@ The following options control enabling TPLs:
     * * ``Kokkos_ENABLE_LIBDL``
       * Whether to enable the LIBDL library
       * ``ON``
+    * * ``Kokkos_ENABLE_ONEDPL``
+      * Whether to enable the oneDPL library
+      * ``ON``
```
Comment (dalg24) [inline docs/source/keywords.rst:230]: No language about it being a SYCL-specific TPL?

**Pull request merged**